assets/js/components/GoogleChartV2.js

```diff
@@ -0,0 +1,174 @@
+
+	const loader = (
+		<div className="googlesitekit-chart-v2-loading">
+			<PreviewBlock
+				className="googlesitekit-chart-v2-loading__wrapper"
+				height={ loadingHeightToUse }
+				shape={ loadingShape }
+				width={ loadingWidthToUse }
+			/>
+		</div>
+	);
+
+	const chartWrapperRef = useRef();
+	const googleRef = useRef();
+
+	useEffect( () => {
+		// Remove all event listeners after the component has unmounted.
+		return () => {
+			// eslint-disable-next-line no-unused-expressions
+			googleRef.current?.visualization.events.removeAllListeners( chartWrapperRef.current?.getChart() );
+		};
+	}, [] );
+
+	// These event listeners are added manually to the current chart because
+	// `react-google-charts` doesn't support `mouseOver` or `mouseOut` events
+	// in its `chartEvents` prop.
+	// See: https://github.com/google/site-kit-wp/pull/2805#discussion_r579172660
+	useLayoutEffect( () => {
+		if ( onMouseOver ) {
+			// eslint-disable-next-line no-unused-expressions
+			googleRef.current?.visualization.events.addListener( chartWrapperRef.current.getChart(), 'onmouseover', ( event ) => {
+				onMouseOver( event, {
+					chartWrapper: chartWrapperRef.current,
+					google: googleRef.current,
+				} );
+			} );
+		}
+
+		if ( onMouseOut ) {
+			// eslint-disable-next-line no-unused-expressions
+			googleRef.current?.visualization.events.addListener( chartWrapperRef.current.getChart(), 'onmouseout', ( event ) => {
+				onMouseOut( event, {
+					chartWrapper: chartWrapperRef.current,
+					google: googleRef.current,
+				} );
+			} );
+		}
+	}, [ onMouseOver, onMouseOut ] );
+
+	if ( ! loaded ) {
+		return (
+			<div className={ classnames(
+				'googlesitekit-chart-v2',
+				'googlesitekit-chart-v2-loading__forced',
+				className
+			) }>
+				{ loader }
+			</div>
+		);
+	}
+
+	const combinedChartEvents = [ ...chartEvents || [] ];
+
+	if ( onReady ) {
+		combinedChartEvents.push( {
+			eventName: 'ready',
+			callback: onReady,
+		} );
+	}
+
+	if ( onSelect ) {
+		combinedChartEvents.push( {
+			eventName: 'select',
+			callback: onSelect,
+		} );
+	}
+
+	return (
+		<Fragment>
+			<Chart
+				chartEvents={ combinedChartEvents }
+				chartType={ chartType }
+				className={ classnames(
+					'googlesitekit-chart-v2',
+					`googlesitekit-chart-v2--${ chartType }`,
+					className
+				) }
+				loader={ loader }
+				height={ height }
+				getChartWrapper={ ( chartWrapper, google ) => {
+					// Remove all the event listeners on the old chart before we draw
+					// a new one. Only run this if the old chart and the new chart aren't
+					// the same reference though, otherwise we'll remove existing `onReady`
+					// events and other event listeners, which will cause bugs.
+					if ( chartWrapper !== chartWrapperRef.current ) {
+						// eslint-disable-next-line no-unused-expressions
+						googleRef.current?.visualization.events.removeAllListeners( chartWrapperRef.current?.getChart() );
```

Comment:
Same here, we also need to remove events assigned to the `chartWrapper` instance.